src/http/Service.cpp

```diff
@@ -146,34 +174,77 @@ QString Service::getKey(const QString &id)
     return QString();
 }
 
+QList<QString> Service::getKeys(const QString &id)
+{
+    QList<QString> keys;
+    QList<Entry*> entries = getConfigEntries();
+    for (Entry* config : entries) {
+        keys << config->attributes()->value(QLatin1String(ASSOCIATE_KEY_PREFIX) + id);
+    }
+    return keys;
+}
+
+//XXX
+//bool Service::attributeExists(QList<Entry*> entries, const QString &attribute)
+bool Service::attributeExists(QList<Entry*> entries, const QString attribute)
```

Comment:
Is `const QString attribute` or `const QString &attribute` correct here?

Reply:
Yes this is correct. The data you allocate as part of the list gets shared with the calling scope's new QList object.